applications/Preprocessor/lib/mesh/MeshImpl/MeshEntity.h

```diff
@@ -173,7 +204,26 @@ class MeshEntity {
     /// \param elementID The entityID of the element
     /// \param localEntityIndex The localIndex of this MeshEntity for the
     /// element.
-    void addElement(EntityGId elementID, EntityLId EntityLId);
+    void addElement(EntityGId elementID, EntityLId localEntityIndex);
+
+    /// Reverse of addElement, remove an Element that this MeshEntity is part
+    /// of.
+    ///
+    /// \param elementId The id of the element to remove
+    /// \param localEntityIndex The local index on the element of this
+    /// MeshEntity (before removal).
+    void removeElement(EntityGId elementId, EntityLId localEntityIndex);
+
+    // Actual implementation of merge depends on the dimension of the
+    // MeshEntity, with the zero dimensional one being slightly different due to
+    // having to deal with coordinates. Partial template specialization is not
+    // possible with member functions, this pair of functions is a poor man's
+    // substitute preventing having to specialize the whole class for just this
+    // function.
+    template <std::size_t d>
+    std::enable_if_t<d == entityDimension> merge(
+        MeshEntity<entityDimension, meshDimension>& source, tag<d>);
+    void merge(MeshEntity<entityDimension, meshDimension>& source, tag<0>);
```

Comment:
@JensWehner Note this is a bit of a hack, if you have any good ideas please let me know.

Reply:
that is fine, limitation of the language.